candump.c

```diff
@@ -635,6 +704,41 @@ int main(int argc, char **argv)
 			continue;
 		}
 
+		// check netlink socket
+		if (FD_ISSET(netlink_s.s, &rdfs)) {
+			char buf[4096];
+			struct iovec iov = { buf, sizeof(buf) };
+			struct msghdr msg = { &netlink_s.saddr, sizeof(netlink_s.saddr), &iov, 1, NULL, 0, 0 };
+
+			int len = recvmsg(netlink_s.s, &msg, 0);
+
+			struct nlmsghdr *nh;
+			for (nh = (struct nlmsghdr *) buf; NLMSG_OK (nh, len); nh = NLMSG_NEXT (nh, len)) {
+				if (nh->nlmsg_type == NLMSG_DONE){
+					break;
+				}
+				if (nh->nlmsg_type == NLMSG_ERROR)
+					continue;
+				if (nh->nlmsg_type == RTM_NEWLINK) {
+					struct ifinfomsg *rtif = NLMSG_DATA(nh);
+
+					if (is_idx_cached(rtif->ifi_index)) {
+						char ifname[IF_NAMESIZE];
+						if_indextoname(rtif->ifi_index, ifname);
+
+						const int will_exit = (down_causes_exit && !(rtif->ifi_flags&IFF_RUNNING));
```

Comment:
please move all variable declarations to the beginning of the scope. (I know, kernel C-style is not consequent in using C99 or not :)